src/harness/reference_models/iap/iap.py

```diff
@@ -459,190 +463,192 @@ def performIapForPpa(protected_entity, sas_uut_fad_object, sas_th_fad_objects,
+      asas_interference, aggr_interference)
 
-def performIapForFssCochannel(protected_entity, sas_uut_fad_object, sas_th_fad_objects):
-  """Compute post IAP interference margin for FSS Co-channel incumbent type
+  return ap_iap_ref
 
-  FSS protection is provided over co-channel pass band
+
+def performIapForFssCochannel(fss_record, sas_uut_fad_object, sas_th_fad_objects):
+  """Computes post IAP interference margin for FSS Co-channel incumbent.
+
+  FSS protection is provided over co-channel pass band.
 
   Args:
-    protected_entity: Protected entity of type FSS co-channel
-    sas_uut_fad_object: FAD object from SAS UUT
-    sas_th_fad_object: a list of FAD objects from SAS Test Harness
-   Returns:
-    ap_iap_ref: post IAP allowed interference in the format of 
-    {latitude : {longitude : [interference(mW/IAPBW), interference(mW/IAPBW)]}}
+    fss_record: A FSS record (dict).
+    sas_uut_fad_object: A FAD object from SAS UUT
+    sas_th_fad_object: A list of FAD objects from SAS Test Harness
+  Returns:
+    ap_iap_ref: The post-IAP |interf.AggregateInterferenceOutputFormat| allowed
+      interference, in the format of:
+      {latitude : {longitude : [interference(mW/IAPBW), .., interference(mW/IAPBW)]}}
   """
-  fss_cochannel_thresh_q = THRESH_FSS_CO_CHANNEL_DBM_PER_RBW\
-                          + 10*np.log10(IAPBW_HZ / FSS_RBW_HZ)
+  fss_cochannel_thresh_q = (THRESH_FSS_CO_CHANNEL_DBM_PER_RBW
+                            + interf.linearToDb(IAPBW_HZ / FSS_RBW_HZ))
+
   # Actual protection threshold used for IAP - calculated by applying
   # a pre-defined Pre-IAP headroom (Mg) at each protection threshold(Q)
-  fss_cochannel_iap_threshold = interf.dbToLinear(fss_cochannel_thresh_q -
-                              MARGIN_FSS_CO_CHANNEL_DB)
+  fss_cochannel_iap_threshold = interf.dbToLinear(fss_cochannel_thresh_q
+                                                  - MARGIN_FSS_CO_CHANNEL_DB)
 
-  grant_objects = data.getGrantObjectsFromFAD(sas_uut_fad_object, sas_th_fad_objects)
+  grants = data.getGrantObjectsFromFAD(sas_uut_fad_object, sas_th_fad_objects)
 
-  # Get number of SAS 
+  # Get number of SAS
   num_sas = len(sas_th_fad_objects) + 1
 
-  # Get the protection point of the FSS
-  fss_point = protected_entity['record']['deploymentParam'][0]['installationParam']
-  protection_point = (fss_point['longitude'], fss_point['latitude'])
-
-  # Get the frequency range of the FSS
-  fss_freq_range = protected_entity['record']['deploymentParam'][0]\
-      ['operationParam']['operationFrequencyRange']
-  fss_low_freq = fss_freq_range['lowFrequency']
-  fss_high_freq = fss_freq_range['highFrequency']
-
-  # Get FSS information
-  fss_info = data.FssInformation(height_agl=fss_point['height'],
-               max_gain_dbi=fss_point['antennaGain'],
-               pointing_azimuth=fss_point['antennaAzimuth'],
-               pointing_elevation=fss_point['antennaDowntilt'])
+  # Get the FSS infos
```

Comment:
used the common/data routine. This also suppress bugs:
 - elevation was wrong (lack the minus)
 - handle the AMSL case